common/djangoapps/oauth_exchange/forms.py

```diff
@@ -0,0 +1,109 @@
+"""
+Forms to support third-party to first-party OAuth 2.0 access token exchange
+"""
+from django.contrib.auth.models import User
+from django.forms import CharField
+from oauth2_provider.constants import SCOPE_NAMES
+import provider.constants
+from provider.forms import OAuthForm, OAuthValidationError
+from provider.oauth2.forms import ScopeChoiceField, ScopeMixin
+from provider.oauth2.models import Client
+from requests import HTTPError
+import social.apps.django_app.utils as social_utils
+from social.backends import oauth as social_oauth
+
+from third_party_auth.provider import Registry
+from third_party_auth import pipeline
+
+
+class AccessTokenExchangeForm(ScopeMixin, OAuthForm):
+    provider = CharField(required=False) # TODO: rename
```

Comment:
As discussed in person but recorded here for posterity, "backend" is a python-social-auth term that will not necessarily make sense to a client of this API, so I went with something that I think is a bit more straightforward.